File: PY/451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        breakdown = {}
        sortable = []
        final = ''
        for char in s:
            if char not in breakdown:
                breakdown[char] = 1
            else :
                breakdown[char] += 1
        sorted_breakdown = sorted(breakdown.items(), key=operator.itemgetter(1))
        sorted_breakdown.reverse()

        for entry in sorted_breakdown:
            entry = [entry[0], entry[1]]
            while entry[1] > 0:
                final += entry[0]
                entry[1] -= 1
        return(final)

File: JS/451-sort-characters-by-frequency.js
var frequencySort = function(s) {
  breakdown = {}
  sortable = []
  final = ''
  arr = s.split('')
  arr.forEach(x => {
    breakdown[x] === undefined ? breakdown[x] = 1 : breakdown[x]++
  })
  for(let x in breakdown){
    sortable.push([x, breakdown[x]])
  }
  sortable.sort((a,b) => {
    return b[1] - a[1]
  })
  sortable.map(x => {
    while(x[1] > 0){
      final+=x[0]
      x[1]--
    }
  })
  return final
};